### ISSUE TITLE
mr.roboto offers to start jenkins jobs for wrong target version

### ISSUE THREAD

**Issue by jensens** [plone/jenkins.plone.org#258]
Jenkins PR jobs for Products.Archetypes on Plone 5.1 are shown for merges into master, but Plone 5.1 uses branch `1.15.x`, so they should not show up for 5.1.
![Screenshot at 2019-04-09 09-34-10](https://user-images.githubusercontent.com/157140/55781539-a58e0900-5aaa-11e9-872c-62d9bcd6d0dd.png)

The buildout fails, for obvious reasons: `masters` dependencies are upgraded for Zope 4:

```
Installing robot.
Version and requirements information containing zexceptions:
  [versions] constraint on zexceptions: 2.13.0
  Requirement of plone.app.event[test]==3.2.3: zExceptions
  Requirement of Zope2: zExceptions
  Requirement of AccessControl>=3.0.0Acquisition: zExceptions
  Requirement of Products.ZCatalog: zExceptions
  Requirement of Zope>=4.0b7.dev0: zExceptions>=3.4
  Requirement of zExceptions: zope.security
  Requirement of zExceptions: zope.publisher
  Requirement of zExceptions: zope.interface
  Requirement of zExceptions: setuptools
  Requirement of DocumentTemplate: zExceptions
  Requirement of Products.ZCTextIndex: zExceptions
  Requirement of Products.PythonScripts: zExceptions
While:
  Installing robot.
Error: The requirement ('zExceptions>=3.4') is not allowed by your [versions] constraint (2.13.0)
```

**Comment by gforcada** [comment]
I will try to have a look

**Comment by mauritsvanrees** [comment]
Same happens for more PRs. This is my latest, for CMFPlone 5.2.x, and it wants a 5.1 job among the other good ones:
https://github.com/plone/Products.CMFPlone/pull/3116

**Comment by gforcada** [comment]
weird :confused: I barely have time, and motivation to begin with, lately, but I will do my best to have a look at it

**Comment by gforcada** [comment]
code is open though, so anyone can have a look

**Comment by mauritsvanrees** [comment]
On https://jenkins.plone.org/roboto/ I see helpful links.  The `sources.json` and `checkouts.json` seem okay, at least for the `CMPlone` repository of my example.

My guess is that the pickle versions of those (especially `sources_json`) are out of sync. So someone needs to click the "Update sources and checkouts" link on that page. But this needs a token that I don't have.

I expect that is is usually done automatically in a cron job, but that something is going wrong there. Or the cron job was accidentally removed, that would be the easiest cause.

**Comment by gforcada** [comment]
I clicked on the link :) rather than a cron job, it was usually updated when mr.roboto detects new changes

**Comment by mauritsvanrees** [comment]
Found it. TLDR: it's not `mr.roboto` or Jenkins.

Longer:

Thanks. Does not seem to help though. I just added PR https://github.com/plone/Products.CMFPlone/pull/3117 for `Products.CMFPlone` branch 5.2.x, and it wants me to run the 5.1 job:

![Screenshot 2020-06-11 at 22 45 47](https://user-images.githubusercontent.com/210587/84437477-57880e80-ac35-11ea-8d54-84e305821bf3.png)

Strangely, the 5.1 line looks slightly different. The others have the Jenkins/mr.roboto logo and status 'Pending'. The 5.1 line has no logo, and status 'Expected'.

Wait a minute. This has nothing to do with `mr.roboto`. On Github we have branch protection rules set up. For the 5.2.x branch it [looks like this](https://github.com/plone/Products.CMFPlone/settings/branch_protection_rules/16148303) (link only for admins):

![Screenshot 2020-06-11 at 22 48 05](https://user-images.githubusercontent.com/210587/84437746-cb2a1b80-ac35-11ea-98e6-b44ded88fd88.png)

I can just remove 5.1 from the required checks there, and it should be fine again.
Done. And... the 5.1 check is immediately gone from the list in the PR. Hurray!

The master branch has the 6.0 jobs and changelog and agreement marked as required.
The 5.1.x and 4.3.x have nothing marked as required.

I wonder if we have scripts running automatically that set this, and might reset it later, or if this was just a manual action by someone.

And I wonder if these required checks in the branch protection rules are really needed, given that mr.roboto does the same, and has an up to date list of where each package branch is used.
